design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,582 @@
+
+### Affected Components
+- Conjur
+
+- DAP
+
+## Security
+* **Future support in Service account key needs to be done very carefully** since there is a concern of impersonate other Conjur hosts, 
+because the Service account key enable you to control the content of the JWT token. 
+The reason it is not a concern in our use case, because we are validating only tokens which supplied by google metadata server, 
+and signed by google private keys which are not expose to the apps in contrast to the Service account key.
+
+## Test Plan
+[//]: # (Fill in the table below to depict the tests that should run to validate your solution)
+[//]: # (You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#)
+
+| **Title** | **Given** | **When** | **Then** | **Comment** |
+|-----------|-----------|----------|----------|-------------|
+|           |           |          |          |             |
+|           |           |          |          |             |
+
+## Logs
+
+### Error Log level
+
+|    | Scenario                                                                | Log message                                                                            | Comment            |
+|--- |-----------------------------------------------------------------------  |----------------------------------------------------------------------------------------|------------------------|
+| 1  | Authenticator is not enabled (in DB/ENV)                                | Authenticator '{0-authenticator-name}' is not enabled                                  | |
+| 2  | Webservice is not defined in a Conjur policy                            | Webservice '{0-webservice-name}' wasn't found                                          | |
+| 3  | Host is not permitted to authenticate with the webservice               | '{0-role-name}' does not have 'authenticate' privilege on {1-service-name}             | |
+| 4  | Host is not defined in Conjur                                           | '{0-role-name}' wasn't found                                                           | |
+| 5  | Couldn't make connection with Google Identity Provider in time          | Google Identity Provider failed with timeout error (Provider URI: '{0}'). Reason: '{1}'| |
+| 6  | Failed to confirm Google token signature                                | Failed to confirm signature of '{0-token}' issued by (Provider URI: '{1}'              | |
+| 7  | Failed to validate Google token claims                                  | Will be decided in Impl phase              | |
+| 8  | Authentication request body is missing a field (e.g `jwt_token`)        | Field '{0-field-name}' is missing or empty in request body                             | |
+| 9  | A required annotation is missing in the Conjur host                     | Will be decided in Impl phase                                                          | Annotation is missing for authentication |
```

Comment:
@nessiLahav please let me review when times come. why is it different from other authenticators?

Reply:
Those message are aligned with Azure, at the development stage you will review them again